### ISSUE TITLE
minitest/hell does not make tests run in parallel

### ISSUE THREAD

**Issue by azul** [seattlerb/minitest#661]
Here's my test script...
```ruby
require 'minitest/autorun'
require 'minitest/pride'
require 'minitest/hell'

class ParallelTest < Minitest::Test
  parallelize_me! if ENV['PARALLELIZE_ME']

  def test_one
    sleep 10
    puts Minitest::VERSION
    assert true
  end

  alias_method :test_two, :test_one
  alias_method :test_three, :test_one

end
```

Here's what i get with parallelize_me!:
```
$ PARALLELIZE_ME=1 ruby test/minitest_hell_test.rb -v
Run options: -v --seed 11032

# Running:

5.10.1
5.10.1
10.00 s = .
10.00 s = .
5.10.1
10.00 s = .

Fabulous run in 20.001489s, 0.1500 runs/s, 0.1500 assertions/s.

3 runs, 3 assertions, 0 failures, 0 errors, 0 skips
```

It should be the same without because of minitest/hell. But here's what i get without parallelize_me!:
```
$ ruby test/minitest_hell_test.rb -v
Run options: -v --seed 14635

# Running:

ParallelTest#test_two = 5.10.1
10.00 s = .
ParallelTest#test_three = 5.10.1
10.00 s = .
ParallelTest#test_one = 5.10.1
10.00 s = .

Fabulous run in 30.001924s, 0.1000 runs/s, 0.1000 assertions/s.

3 runs, 3 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by azul** [comment]
removing minitest/pride does not make a difference.

**Comment by azul** [comment]
`self.class.test_order` is `:parallel` in both cases. It just does not seem to have any effect with minitest/hell.

**Comment by azul** [comment]
The part that is missing seems to be in
```ruby
extend Minitest::Parallel::Test::ClassMethods
```

**Comment by azul** [comment]
Looks like this is what's missing from minitest hell:
```
  def run_one_method klass, method_name, reporter
    Minitest.parallel_executor << [klass, method_name, reporter]
  end
```
Adding it to the test makes it run in parallel.

**Comment by azul** [comment]
Seems to be a leftover from https://github.com/seattlerb/minitest/commit/34760e3b268bc1bb4ac5fe1a44ef1ff0a2f9bd4d

**Comment by azul** [comment]
I have tried to come up with a fix and a test for this... But it's hard to get working without breaking the current test suite by plain requiring `minitest/hell`. I suggest dropping it entirely. The same effect can be achieved with
```ruby
class Minitest::Test
  paralellize_me!
end
```

**Comment by azul** [comment]
Tried to fix it in #662 - have difficulties with the tests still as requiring minitest/hell messes up the other tests.

**Comment by zenspider** [comment]
Fixed! Thanks!
